src/components/RefineSearch/RefineSearch.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Box,
+  Button,
+  HorizontalRule,
+  ButtonGroup,
+  Form,
+} from "@nypl/design-system-react-components"
+import type { Dispatch, SyntheticEvent } from "react"
+import { useState, useRef, useCallback } from "react"
+import { useRouter } from "next/router"
+
+import styles from "../../../styles/components/Search.module.scss"
+import SearchResultsFilters from "../../models/SearchResultsFilters"
+import RefineSearchCheckBoxField from "./RefineSearchCheckboxField"
+import {
+  collapseMultiValueQueryParams,
+  buildFilterQuery,
+  getQueryWithoutFilters,
+} from "../../utils/refineSearchUtils"
+import type { Aggregation } from "../../types/filterTypes"
+
+interface RefineSearchProps {
+  aggregations: Aggregation[]
+  setAppliedFilters: Dispatch<React.SetStateAction<Record<string, string[]>>>
+  appliedFilters: Record<string, string[]>
+}
+
+const RefineSearch = ({
+  aggregations,
+  appliedFilters,
+  setAppliedFilters,
+}: RefineSearchProps) => {
+  const router = useRouter()
+  const fields = useRef([
```

Comment:
I added a bunch of useref/usecallback etc wrappers when I was trying to eliminate some extra rendering thinking it might make the lag go away. Are you saying I should declare this as a constant outside of the component?

Reply:
Whoops yes, my comment wasn't too clear. Using `useRef` or `useCallback` does keep the same value between rererenders but if you define the array outside of the component, it's a similar behavior because the component doesn't have to define that array on every rerender.

Here's an interesting take https://stackoverflow.com/questions/62848106/functional-component-write-functions-inside-or-outside-the-component

I agree that it's a premature optimization and in this case, it's an array with a few objects which is "cheap" to recreate on every render. So if the entire feature is laggy it may be related to something else.

Reply:
In this case, I mentioned this mostly to not use `useRef`. Not that it's a bad function to use but if you can prevent its use that means one less react function to import.